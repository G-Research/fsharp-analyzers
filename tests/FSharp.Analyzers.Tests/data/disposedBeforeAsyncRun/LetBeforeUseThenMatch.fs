namespace N

open System

type DisposableThing () =
    interface IDisposable with
        member _.Dispose() = ()

[<RequireQualifiedAccess>]
module DisposableThing =
    let make () =
        new DisposableThing (), 5

module Program =
    let pushFiles () =
        let blah, anInt = DisposableThing.make ()
        use blah = blah
        match anInt with
        | 3 -> failwith "it wasn't 5"
        | _ ->
            async {
                return "hi"
            }
