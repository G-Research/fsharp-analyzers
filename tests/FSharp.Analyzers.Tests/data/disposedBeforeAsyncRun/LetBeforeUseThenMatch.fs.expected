GRA-DISPBEFOREASYNC-001 | Warning | (17,8 - 17,16) | Object is disposed before returned async is run | []
